src/Infrastructure/Dashboard/TermMeta/TermMetaRegistrar.php

```diff
@@ -64,21 +72,83 @@ function () use ($termMetaFields): void {
 
         add_action("edit_{$taxonomy}", $updateCallback);
         add_action("create_{$taxonomy}", $updateCallback);
+        add_filter('pre_insert_term', $this->buildValidationCallback(...$termMetaFields));
+    }
+
+    // phpcs:ignore Inpsyde.CodeQuality.NestingLevel.High
+    private function buildValidationCallback(TermMetaField ...$termMetaFields): callable
+    {
+        return function (mixed $term) use ($termMetaFields): mixed {
+            if ($term instanceof WP_Error) {
+                return $term;
+            }
+
+            $error = null;
+
+            foreach ($termMetaFields as $termMetaField) {
+                $postedValue = filter_input(
+                    INPUT_POST,
+                    $termMetaField->key(),
+                    FILTER_SANITIZE_SPECIAL_CHARS
+                );
+
+                $error = $this->checkForErrors($termMetaField, $postedValue);
+
+                if ($error instanceof WP_Error) {
+                    return $error;
+                }
+            }
+
+            return $term;
+        };
+    }
+
+    private function checkForErrors(TermMetaField $termMetaField, mixed $value): ?WP_Error
+    {
+        foreach ($termMetaField->validationRules() as $validationRule) {
+            if ($validationRule->passes($value)) {
+                continue;
+            }
+
+            return new WP_Error(
+                'invalid_term_meta',
+                $this->validationMessages->get($validationRule, $termMetaField->title()),
+            );
+        }
+
+        return null;
     }
 
     private function buildUpdateCallback(TermMetaField ...$termMetaFields): callable
     {
         return function (int $termId) use ($termMetaFields): void {
             foreach ($termMetaFields as $termMetaField) {
-                // phpcs:ignore WordPressVIPMinimum.Security.PHPFilterFunctions.MissingThirdParameter
                 $postedValue = filter_input(
                     INPUT_POST,
                     $termMetaField->key(),
+                    FILTER_SANITIZE_SPECIAL_CHARS
```

Comment:
I have a couple of reservations for this change. I 'm quite sure that there was (is?) a pretty good reason for the filter param missing. Are all term meta fields expected to have just plain strings (i.e. no HTML formatting etc)?